docs/dep/dep-006-new-go-sdk.md

```diff
@@ -0,0 +1,474 @@
+	}
+	
+	return nil
+}
+```
+
+The code above is just an example; the idea is that `Run` will execute the `Runner` actions
+and provide for free a lot of functionalities that are needed to build reliable, debuggable and efficient components.
+
+The actions executed by run will be configurable and have notions of retry policies and custom error handling.
+
+*Not all of these functionalities will be provided from day one. They'll be added one at a time.*
+
+<a id="proposal-component-api-defaults"></a>
+
+#### Defaults
+
+The component SDK is designed to provide overridable smart defaults to provide reliability, monitoring and debugging
+capabilities for free to adopters.
+
+These functionalities will be shared across actions and configured in `Config`.
+
+We won't cover them all here. Just to give an example, considering logging and panic handling:
+
+```go
+type (
+    // PanicHandler defines a generic contract for handling panics following the recover semantics.
+    PanicHandler interface {
+        HandlePanic()
+    }
+
+    // Logger exposes an slog.Logger compatible logger contract.
+    Logger interface {
+        Debug(msg string, keyvals ...any)
+        Info(msg string, keyvals ...any)
+        Warn(msg string, keyvals ...any)
+        Error(msg string, keyvals ...any)
+    }
+    
+    Config struct {
+        Logger       Logger
+        PanicHandler PanicHandler
+    }
+
+    // ConfigOption can be used to override configuration defaults.
+    // For example overriding the default logger.
+    ConfigOption func(*Config) error
+	
+    defaultPanicHandler struct {
+        logger Logger
+    }
+)
+
+func (dph defaultPanicHandler) HandlePanic() {
+    if err := recover(); err != nil {
+        dph.logger.Error("recovered from panic: %v", err)
+    }
+}
+
+// NewConfig returns a new configuration with overridable defaults.
+func NewConfig(opts ConfigOption...) (*Config, error) {
+    logger := Logger: log.New(slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{
+        Level: slog.LevelDebug,
+    }))
+	
+    cfg := &Config{
+        Logger: logger,
+        PanicHandler: defaultPanicHandler{logger: logger},
+    }
+
+    for _, opt := range opts {
+        if err := opt(cfg); err != nil { ... }
+    }
+	
+    return cfg, nil
+}
+```
+
+<a id="proposal-component-api-example"></a>
+
+### Example
+
+Essentially, one has to just implement a `Runner` to create a new component and that will be it.
+
+<details>
+  <summary>Simple Example</summary>
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/ocurity/dracon/sdk/component"
+	"github.com/ocurity/dracon/api/gen/com/github/ocsf/ocsf-schema/v1"
+)
+
+// simpleComponent parses a Vulnerability Findings report from JSON
+// enriches it and stores the processed output to SQLite.
+type simpleComponent struct{}
+
+func (s simpleComponent) Read(ctx context.Context) (*v1.VulnerabilityFinding, error) {
+	b, err := os.ReadFile("input.json")
```

Comment:
Mirroring offline conversation:
Can we please abstract the read to the sdk? with this implementation, if we ever change where we read things from we need to rewrite every component to match, if Read just has to call a "Give me data" we can then easily abstract where we read data from to the sdk and make it a configuration details

Reply:
Yes, will extend the doc with this section :)